packages/liveblocks-node/src/client.ts

```diff
@@ -194,6 +273,512 @@ export class Liveblocks {
     }
   }
 
+  /* -------------------------------------------------------------------------------------------------
+   * Room
+   * -----------------------------------------------------------------------------------------------*/
+
+  /**
+   * Returns a list of your rooms. The rooms are returned sorted by creation date, from newest to oldest. You can filter rooms by metadata, users accesses and groups accesses.
+   * @param params.limit (optional) A limit on the number of rooms to be returned. The limit can range between 1 and 100, and defaults to 20.
+   * @param params.startingAfter (optional) A cursor used for pagination. You get the value from the response of the previous page.
+   * @param params.userId A filter on users accesses.
+   * @param params.metadata A filter on metadata. Multiple metadata keys can be used to filter rooms.
+   * @param params.groupIds A filter on groups accesses. Multiple groups can be used.
+   * @returns A list of rooms.
+   */
+  public async getRooms(params: {
+    limit?: number;
+    startingAfter?: number;
+    metadata: RoomMetadata;
+    userId: string;
+    groupIds: string;
+  }): Promise<{
+    nextPage: string | null;
+    data: RoomInfo[];
+  }> {
+    let path = "v2/rooms?";
+    if (params.limit) {
+      path += `&limit=${params.limit}`;
+    }
+
+    if (params.startingAfter) {
+      path += `&startingAfter=${params.startingAfter}`;
+    }
+
+    if (params.userId) {
+      path += `&userId=${params.userId}`;
+    }
+
+    if (params.groupIds) {
+      path += `&groupIds=${params.groupIds}`;
+    }
+
+    if (params.metadata) {
+      Object.entries(params.metadata).forEach(([key, val]) => {
+        path += `&metadata.${key}=${val}`;
+      });
+    }
+
+    const res = await this.get(`/${path}`);
+
+    if (!res.ok) {
+      const text = await res.text();
+      throw new ApiError(res.status, text);
+    }
+
+    return res.json();
```

Comment:
```suggestion
    return await res.json();
```

Reply:
I am not sure if this makes any difference in terms of the function's behavior or the return type. That being said, I understand that using await can sometimes make error handling clearer and stack traces more accurate. @nvie Do you have a preference?

Reply:
Ahhh yes. Originally, we followed the ESLint rule `no-return-await` for this — so we were deliberately avoiding the `return await` construct, because it would be less performant. (Return-await would create a new promise, which would only resolve in the next tick. So basically a trade-off of stack trace info for a bit more performance. And, of course, code consistency.)

I just read up a bit more on the latest state of this, and it looks like V8 recently added an optimization for this, making the `return await promise` _faster_ than `return promise`, so the default nowadays is the best of both worlds: more performance _and_ better stack traces.

<img width="885" alt="Screen Shot 2023-11-06 at 10 28 28@2x" src="https://github.com/liveblocks/liveblocks/assets/83844/61c3b8b9-b676-4f24-bdd1-f8c843b85423">

I think this means we should start discouraging `return promise` in favor of `return await promise` indeed, so you're making the right call here! Let's do a pass where we change our default inside `shared/eslint-config` (and in our other repos), and change this across the board in a follow-up PR.

Reply:
This is very interesting! Thank you for looking into this.